packages/jazz-react/src/provider.tsx

```diff
@@ -1,117 +1,204 @@
 import {
   BaseBrowserContextOptions,
+  BrowserContext,
+  BrowserGuestContext,
   createJazzBrowserContext,
 } from "jazz-browser";
-import React, { useEffect, useRef, useState } from "react";
+import React, { useEffect } from "react";
 
 import { JazzContext, JazzContextType } from "jazz-react-core";
 import { Account, AccountClass, AuthMethod } from "jazz-tools";
+import { useIsAnonymousUser } from "./auth/AnonymousAuth.js";
 
 export interface Register {}
 
 export type RegisteredAccount = Register extends { Account: infer Acc }
   ? Acc
   : Account;
 
-export type JazzProviderProps<Acc extends Account = RegisteredAccount> = {
-  children: React.ReactNode;
-  auth: AuthMethod | "guest";
+export type JazzContextManagerProps<Acc extends Account = RegisteredAccount> = {
+  auth?: AuthMethod | "guest";
   peer: `wss://${string}` | `ws://${string}`;
+  localOnly?: boolean;
   storage?: BaseBrowserContextOptions["storage"];
   AccountSchema?: AccountClass<Acc>;
 };
 
+export type JazzProviderProps<Acc extends Account = RegisteredAccount> = {
+  children: React.ReactNode;
+  localOnly?: boolean | "anonymous";
+} & Omit<JazzContextManagerProps<Acc>, "localOnly">;
+
+class JazzContextManager<Acc extends Account = RegisteredAccount> {
```

Comment:
Refactored to use `useSyncExternalStorage` and use a "propsChanged" check to avoid the double invocation on dev